stonesoup/updater/kalman.py

```diff
@@ -901,3 +902,133 @@ def predict_measurement(self, predicted_state, measurement_model=None, measureme
 
         return MeasurementPrediction.from_state(
             predicted_state, meas_pred_mean, meas_pred_covar, cross_covar=cross_covar)
+
+
+class StochasticIntegrationUpdater(KalmanUpdater):
+    """Stochastic Integration Kalman Filter class. Inherits most
+    of the functionality from :class:`~.KalmanUpdater`.
+
+    The measurement update of nonlinear measurement models is accomplished by
+    the stochastic integration approximation.
+
+    """
+
+    # Can be non-linear and non-differentiable
+    measurement_model: MeasurementModel = Property(
+        default=None,
+        doc="The measurement model to be used. This need not be defined if a "
+        "measurement model is provided in the measurement. If no model "
+        "specified on construction, or in the measurement, then error "
+        "will be thrown.",
+    )
+    Nmax: int = Property(default=10, doc="maximal number of iterations of SIR")
+    Nmin: int = Property(
+        default=5,
+        doc="minimal number of iterations of stochastic integration rule (SIR)",
+    )
+    Eps: float = Property(default=5e-4, doc="allowed threshold for integration error")
+    SIorder: int = Property(
+        default=5, doc="order of SIR (orders 1, 3, 5 are currently supported)"
+    )
+
+    @lru_cache()
+    def predict_measurement(
+        self, predicted_state, measurement_model=None, measurement_noise=True, **kwargs
+    ):
+        """Stochastic Integration Filter measurement prediction step. Uses
+        stochastic integration to estimate a Gaussian distributed predicted measurement.
+
+        Parameters
+        ----------
+        predicted_state : :class:`~.GaussianStatePrediction`
+            A predicted state
+        measurement_model : :class:`~.MeasurementModel`, optional
+            The measurement model used to generate the measurement prediction.
+            This should be used in cases where the measurement model is
+            dependent on the received measurement (the default is `None`, in
+            which case the updater will use the measurement model specified on
+            initialisation)
+
```

Comment:
documentation for measurement_noise, like: 
```suggestion
        predicted_state : :class:`~.GaussianStatePrediction`
            A predicted state
        measurement_model : :class:`~.MeasurementModel`, optional
            The measurement model used to generate the measurement prediction.
            This should be used in cases where the measurement model is
            dependent on the received measurement (the default is `None`, in
            which case the updater will use the measurement model specified on
            initialisation)
        measurement_noise : bool
            Include measurement noise or not

```